cmd/gateway/zerostor/errors/warn.go

```diff
@@ -0,0 +1,38 @@
+package errors
+
+type causer interface {
+	Cause() error
+}
+
+//Warning is a wrapper that demotes an error to warning level
+type warning struct {
+	error
+}
+
+//Cause gets the cause of the warning
+func (w warning) Cause() error {
+	return w.error
+}
+
+//Warning marks this error as a warning
+func Warning(err error) error {
+	if _, ok := err.(warning); ok {
+		return err
+	}
+
+	return warning{err}
```

Comment:
wanring doesn't implement the Error interface, how can you return it here since the function must return an `error`

Reply:
well, obviously this works otherwise i won't create a PR. 
the explantion is, warning componese an error like 
```go
type warning struct{
   error
}
```
not
```go
type warning struct{
  err error
}
```

Reply:
my bad